detection_rules/config.py

```diff
@@ -156,10 +157,13 @@ def parse_rules_config(path: Optional[Path] = None) -> RulesConfig:
     if path:
         assert path.exists(), f'rules config file does not exist: {path}'
         loaded = yaml.safe_load(path.read_text())
-    elif CUSTOM_RULES_DIR:
-        path = Path(CUSTOM_RULES_DIR) / '_config.yaml'
-        assert path.exists(), f'_config.yaml file missing in {CUSTOM_RULES_DIR}'
+    elif CUSTOM_RULES_DIR and (path := Path(CUSTOM_RULES_DIR) / '_config.yaml').exists():
         loaded = yaml.safe_load(path.read_text())
+        if 'configuration_details' in loaded:
+            print(loaded['configuration_details'])
+            path.unlink()
+            sys.exit(0)
```

Comment:
It's probably worth noting that if the config was implemented with a lazy load option, it probably would save us some trouble.

Reply:
>Not showing an entire stack trace which can be misinterpreted as broken code by unexperienced users.

If this is your concern, catch it from the caller (CLI) and print a more specific, slimmed down error (or even `ClientError`)

Reply:
I think the only issue like I mentioned is that the existing logic calls `parse_rules_config` as a global variable in some init files, which I thought occurred earlier in execution.

Reply:
While I understand the hesitation of using sys.exit for handling errors, it is a clean way (and my understanding it is the preferred clean way) of exiting a python program where there is no error. Also, for the case of preferring ctx.exit I agree, when this is available we should use it; however, I do not think we would yet be in the context of click. 

As it is written I think the `parse_rules_config` would load during the imports prior to the establishing of the click context, making for either confusing code or potentially a failed call. It may appear to function but the desired behavior of gracefully terminating from the context may not be available. 

For a graceful, non-error termination sys.exit wraps a call to `raise SystemExit`. While we could directly use `raise SystemExit` this could also be confusing as raise is typically reserved for exceptions and this would imply that we are raising due to an error, which we are not. Functionally, `raise SystemExit` and sys.exit are in effect identical, so that is more of a stylistic preference, but I would err on the side of being clear that this is not an error.

Reply:
we do not track specific error codes to types of errors. An exception will result in python exiting with a non-zero, achieving the desired result. Wrapping in `sys.exit` does nothing but hide the stack trace as you intended, however, I think that is the better thing to do here - let is raise. The error string is obvious enough, and it points users to where in the code it is erroring if they want to understand.

Otherwise, we can force the parse_config to load predictably (since we _can_ lazy load it) and wrap it in an exception there

Reply:
If we intend to lazy load, let's do it outside of this PR. Outside of making adding exception handling to `parse_rules_config` calls, im more inclined to leave this as-is.

Reply:
> it looks like it is `detection-rules/schemas/__init__.py`
> 
> if we add this to `__main__` we can reliably control the error
> 
> ```python
> try:
>     RULES_CONFIG = parse_rules_config()
> except SpecificErrorRaised as e:
>     print(pretty error, sys.stderr)
>     raise # (or sys.exit)
> ```

Let's do that. Not the best, but IMO better than the alternatives.

Reply:
based on how the repo operates, we really _can't_ lazy load. I think this is the best way to do it. The question still remains though - raise exception or print and sys.exit

Reply:
We could call try/except everywhere that uses calls this function when setting a global variable, but it really starts looking messy.

Reply:
![Screenshot 2024-05-07 at 4 50 59 PM](https://github.com/elastic/detection-rules/assets/1636709/95eed14b-5f7a-4297-ae1e-8b744bfee557)

Unfortunately this was added like this in 8 different places so im not sure if try/except is the way to go.

Reply:
The only other thing remaining 3653 i think is this. TLDR; assert with stack trace, or exit clean with sys.exit. IMO adding try/excepts around RULES_CONFIG = parse_rules_config() throughout the 8 global variables in the repo isn't the best solution. If there are no other ideas, I'm inclined to leave the `sys.exit` line in, since it potentially provides a cleaner experience for users.

Reply:
I agree :+1:  I think that the cleanest method of dealing with this situation is to use `sys.exit` with the `0` argument (or no argument at all as the default is `0`).

Reply:
🚀 Using @eric-forte-elastic [approach](https://github.com/elastic/detection-rules/pull/3653#discussion_r1593092457), I think is a happy medium so I rolled that in!